src/org/ods/orchestration/usecase/BitbucketTraceabilityUseCase.groovy

```diff
@@ -37,109 +32,41 @@ class BitbucketTraceabilityUseCase {
      * @return absolutePath of the created file
      */
     List<Map> getPRMergeInfo() {
-        String csvFIleWithInfo = generateSourceCodeReviewFile()
-        return readSourceCodeReviewFile(csvFIleWithInfo)
-    }
-
-    /**
-     * Create a CSV file that contains the following records
-     * for every merge event into the integration branch of every ODS component:
-     * @return absolutePath of the created file
-     */
-
-    String generateSourceCodeReviewFile() {
-        String token = bitbucketService.getToken()
-        File file = createReportFile()
-        processRepositories(file, token)
-        return file.absolutePath
-    }
-
-    /**
-     * Read an existing csv file and parse the info to obtaining an structured List of data.
-     *
-     * @param filePath The csv
-     * @return List of commits
-     */
-    @SuppressWarnings(['JavaIoPackageAccess'])
-    @NonCPS
-    List<Map> readSourceCodeReviewFile(String filePath) {
-        def file = new File(filePath)
-        def result = []
-        def data = processCsv(
-            file.text,
-            Record.CSV,
-            ['commitDate', 'author', 'reviewers', 'pullRequestUrl', 'commitSHA', 'component']
-        ).findAll { it != null }
-
-        def dataSize = data.size()
-        for (def i = 0; i < dataSize; i++) {
-            def info = data[i]
-            result << getCommitInfo(info)
+        def mergeInfo = processRepositories().collect { record ->
```

Comment:
Please can you change it by a "for" with index. We saw it's better for the nonSerialException

Reply:
Actually, I forgot checking the @NonCPS annotations and there's a mistake. I will fix it right away.

Reply:
Now the token is correctly obtained from CPS code and the rest of the processing has been made NonCPS, so there will be no NonSerializableException issues.

Reply:
As per the conversation maintained, I will change it to an indexed loop anyway.